svo (setup, misc, empty, funnies, dor).xml

```diff
@@ -386,128 +389,655 @@ do
+  signals.gmcpcharvitals:connect(function()
+    if gmcp.Char.Vitals.charstats then
+      for _, val in ipairs(gmcp.Char.Vitals.charstats) do
+        local word = val:match("^Word: (%w+)$")
+        if word and word == "Yes" then
+          svo.bals.word = true
+        elseif word and word == "No" then
+          svo.bals.word = false
+          break
+        end
+      end
+    end
+    if not svo.bals.word then
+      svo.bals.word = true
+    end
+  end, 'update word balance')
+  signals.gmcpcharvitals:connect(function()
+    if gmcp.Char.Vitals.charstats then
+      for _, val in ipairs(gmcp.Char.Vitals.charstats) do
+        local age = val:match("^Age: (%d+)$")
+        if age then
+         svo.classinfo.age = tonumber(age)
+          break
+        end
+      end
+    end
+    if not svo.classinfo.age then
+      svo.classinfo.age = 0
+    end
+  end, 'update age')
+end
+
+--Druid
+if svo.me.class == "Druid" then
+  signals.gmcpcharvitals:connect(function()
+    if gmcp.Char.Vitals.charstats then
+      for _, val in ipairs(gmcp.Char.Vitals.charstats) do
+        local sunlight = val:match("^Sunlight: (%d+)$")
+        if sunlight then
+          svo.classinfo.sunlight = tonumber(sunlight)
+          break
+        end
+      end
+    end
+    if not svo.classinfo.sunlight then
+      svo.classinfo.sunlight = 0
+    end
+  end, 'update sunlight')
+  signals.gmcpcharvitals:connect(function()
+    if gmcp.Char.Vitals.charstats then
+      for _, val in ipairs(gmcp.Char.Vitals.charstats) do
+        local morph = val:match("^Morph: (%w+)$")
+        if morph then
+          morph = morph:lower()
+          svo.classinfo.morph = morph
+          if not defc[morph] then
+            sk.clearmorphs()
+            if morph ~= 'none' then
+              defences.got(morph)
+            end
           end
+          break
         end
-        break
       end
     end
-  end
-  if not me.morph then
-    me.morph = ""
-  end
-end, 'update morph')
+    if not svo.classinfo.morph then
+      svo.classinfo.morph = ""
+    end
+  end, 'update morph')
+end
+
+--Infernal
+if svo.me.class == "Infernal" then
+  signals.gmcpcharvitals:connect(function()
+    if gmcp.Char.Vitals.charstats then
+      for _, val in ipairs(gmcp.Char.Vitals.charstats) do
+        local essence = val:match("^Essence: (%w+)$")
+        if essence then
+          svo.classinfo.essence = tonumber(essence)
+          break
+        end
+      end
+    end
+    if not svo.classinfo.essence then
+      svo.classinfo.essence = 0
+    end
+  end, 'update essence')
+  signals.gmcpcharvitals:connect(function()
+    if gmcp.Char.Vitals.charstats then
+      for _, val in ipairs(gmcp.Char.Vitals.charstats) do
+        local spec = val:match("^Spec: (.*)$")
+        if spec then
+          svo.classinfo.spec = spec
+        end
+      end
+    end
+    if not svo.classinfo.spec then
+       svo.classinfo.spec = "None"
+    end
+  end, 'update knightspec')
```

Comment:
I'm not sure if this is a typo or on purpose. 'update knightspec' vs 'update knight spec'.

All the knights have the the same so may not matter.

Sidenote: Achaea refers to these classes as 'warriors' now if that matters.